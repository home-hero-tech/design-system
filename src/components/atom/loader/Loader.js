import React from 'react';
import PropTypes from 'prop-types';

import css from './Loader.module.scss';

const Label = ({ text }) => (
  <div className={css['c-loader']}>
    <div className={css.spinner} />
    <p>{text}</p>
  </div>
);

Label.propTypes = {
  text: PropTypes.string
};

Label.defaultProps = {
  text: 'Carregando'
};

export default Label;
