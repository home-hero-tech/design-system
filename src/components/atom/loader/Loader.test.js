import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loader from './Loader';

describe('Loader', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(<Loader text="Loading" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
