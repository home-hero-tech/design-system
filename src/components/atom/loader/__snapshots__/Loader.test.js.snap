// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match with the snapshot 1`] = `
<div
  className="c-loader"
>
  <div
    className="spinner"
  />
  <Text
    center={false}
    danger={false}
    end={false}
    fill={false}
    lg={false}
    primary={false}
    sm={false}
    success={false}
    uppercase={false}
    warning={false}
    xs={false}
  >
    Loading
  </Text>
</div>
`;
